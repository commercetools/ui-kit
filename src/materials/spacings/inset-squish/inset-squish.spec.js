import React from 'react';
import { shallow } from 'enzyme';
import InsetSquish from './inset-squish';

describe('rendering', () => {
  describe('by default', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <InsetSquish>
          <div />
        </InsetSquish>
      );
    });

    it('should output correc tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should apply `m` as the scale class name', () => {
      expect(wrapper).toHaveClassName('m');
    });

    it('should render `children`', () => {
      expect(wrapper.children()).toBeDefined();
    });
  });

  describe('with `scale`', () => {
    let wrapper;
    let props;

    beforeEach(() => {
      props = { scale: 'l' };
      wrapper = shallow(
        <InsetSquish {...props}>
          <div />
        </InsetSquish>
      );
    });

    it('should output correc tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should apply the `scale` class name', () => {
      expect(wrapper).toHaveClassName(props.scale);
    });
  });
});
