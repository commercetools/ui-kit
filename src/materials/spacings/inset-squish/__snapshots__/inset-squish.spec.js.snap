// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering by default should output correc tree 1`] = `
<div
  className="m"
>
  <div />
</div>
`;

exports[`rendering with \`scale\` should output correc tree 1`] = `
<div
  className="l"
>
  <div />
</div>
`;
